testutil: add ErrorIs test checker

MiguelPires opened the pull request:
This PR adds a new test checker that will be used to replaced error equality checks in tests for their `ErrorIs` equivalent. The idea is that some of the usages of `ErrNoState` don't have all the relevant information and some should even have entirely different error messages. Before we can replace `ErrNoState` where it's returned, we want to change the places that consume it to use `errors.Is` so that we can then change the return sites gradually. Ticket with context: https://warthogs.atlassian.net/browse/SNAPDENG-120
The next PR will replace the equals checker with this one, in the appropriate test checks, as well as occurrences of `!=` and `==` with the `errors.Is(...)` equivalent.

pedronis commented:
this was not the actual plan I discussed with @mvo

mvo5 commented on testutil/errorischecker.go:
```diff
@@ -0,0 +1,35 @@
+
+// ErrorIs calls errors.Is with the provided arguments.
+var ErrorIs = &errorIsChecker{
+	&check.CheckerInfo{Name: "ErrorIs", Params: []string{"error", "target"}},
```
:heart:

mvo5 reviewed (approved):
Thanks, looks fine

pedronis reviewed (changes requested):
this is starting attacking the problem from the wrong side, the issue we have is also that some places abuse/reuse ErrNoState so simply swapping it everywhere with a struct is not great either

pedronis commented:
@MiguelPires the checker is good though, it can be introduced together with the changes to errors.Is

MiguelPires commented:
@pedronis I changed this PR to just introduce the test checker and removed the error struct, as we discussed. The next PR will do the actual replacing of `!=` or `Equals` in the "consumer" sites.
